airflow/api_fastapi/core_api/security.py

```diff
@@ -75,23 +83,63 @@ async def get_user_with_exception_handling(request: Request) -> BaseUser | None:
     return get_user(token_str)
 
 
-def requires_access_dag(method: ResourceMethod, access_entity: DagAccessEntity | None = None) -> Callable:
+def requires_access_dag(
+    method: ResourceMethod, access_entity: DagAccessEntity | None = None
+) -> Callable[[Request, BaseUser | None], None]:
     def inner(
-        dag_id: str | None = None,
+        request: Request,
         user: Annotated[BaseUser | None, Depends(get_user)] = None,
     ) -> None:
-        def callback():
-            return get_auth_manager().is_authorized_dag(
-                method=method, access_entity=access_entity, details=DagDetails(id=dag_id), user=user
-            )
+        dag_id: str | None = request.path_params.get("dag_id")
 
         _requires_access(
-            is_authorized_callback=callback,
+            is_authorized_callback=lambda: get_auth_manager().is_authorized_dag(
+                method=method, access_entity=access_entity, details=DagDetails(id=dag_id), user=user
+            )
         )
 
     return inner
 
 
+class PermittedDagFilter(BaseParam[set[str]]):
+    """A parameter that filters the permitted dags for the user."""
+
+    @classmethod
+    def depends(cls, *args: Any, **kwargs: Any) -> Self:
+        raise NotImplementedError("Use permitted_dag_filter_factory instead , depends is not implemented.")
+
+    def to_orm(self, select: Select) -> Select:
+        from airflow.models.dag import DagModel
```

Comment:
Not necessary. DagModel is already imported because we load the `dag_router` and a lot of other places (param) where this is imported.